test/unit-test/ota_utest.c

```diff
@@ -1430,6 +1460,28 @@ void test_OTA_ProcessJobDocumentNumOverflow()
     TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
 }
 
+void test_OTA_ProcessCustomJobAppCallbackFails()
+{
+    pOtaJobDoc = JOB_DOC_MISSING_KEY;
+    otaInit( pOtaDefaultClientId, mockAppCallbackCustomParsingFails );
+    TEST_ASSERT_EQUAL( OtaAgentStateInit, OTA_GetState() );
+
+    otaGoToState( OtaAgentStateCreatingFile );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+}
+
+void test_OTA_ProcessCustomJobMissingKeys()
+{
+    pOtaJobDoc = JOB_DOC_MISSING_JOB_DOC;
+
+    otaGoToState( OtaAgentStateCreatingFile );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+
+    pOtaJobDoc = JOB_DOC_MISSING_JOB_ID;
+    otaGoToState( OtaAgentStateCreatingFile );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+}
+
```

Comment:
This test is kind of confusing to me. It's called `test_OTA_ProcessCustomJobMissingKeys` but it's using the job documents `JOB_DOC_MISSING_JOB_DOC` and `JOB_DOC_MISSING_JOB_ID`. Based on the name, I would have expected this to use the `JOB_DOC_MISSING_KEY` document.

Reply:
For a custom job, we have a check for only 2 keys: `jobid` and `jobdocument`. Other job  parsing tests are prefixed with 'ProcessJobDocument' to show the distinction.